riposte-async-http-client/src/test/java/com/nike/riposte/client/asynchttp/ning/RequestBuilderWrapperTest.java

```diff
@@ -0,0 +1,150 @@
+package com.nike.riposte.client.asynchttp.ning;
+
+import com.nike.fastbreak.CircuitBreaker;
+import com.nike.fastbreak.CircuitBreakerImpl;
+import com.ning.http.client.AsyncHttpClient;
+import com.ning.http.client.Response;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.http.HttpMethod;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+
+public class RequestBuilderWrapperTest {
+
+    private RequestBuilderWrapper requestBuilderWrapper;
+    private String url;
+    private String httpMethod;
+    private AsyncHttpClient.BoundRequestBuilder requestBuilder;
+    private Optional<CircuitBreaker<Response>> customCircuitBreaker;
+    private boolean disableCircuitBreaker;
+
+    @Before
+    public void setup() {
+        url = "http://localhost.com";
+        httpMethod = HttpMethod.GET.name();
+        requestBuilder = mock(AsyncHttpClient.BoundRequestBuilder.class);
+        customCircuitBreaker = Optional.of(new CircuitBreakerImpl<Response>());
+        disableCircuitBreaker = true;
+    }
+
+    @Test
+    public void constructor_sets_values_as_expected() {
+        // given
+        requestBuilderWrapper = new RequestBuilderWrapper(
+                url,
+                httpMethod,
+                requestBuilder,
+                customCircuitBreaker,
+                disableCircuitBreaker);
+
+        // then
+        assertThat(requestBuilderWrapper.url).isEqualTo(url);
+        assertThat(requestBuilderWrapper.httpMethod).isEqualTo(httpMethod);
+        assertThat(requestBuilderWrapper.customCircuitBreaker).isEqualTo(customCircuitBreaker);
+        assertThat(requestBuilderWrapper.disableCircuitBreaker).isEqualTo(disableCircuitBreaker);
+    }
+
+    @Test
+    public void get_set_ChannelHandlerContext_works_as_expected() {
+        // given
+        requestBuilderWrapper = new RequestBuilderWrapper(
+                url,
+                httpMethod,
+                requestBuilder,
+                customCircuitBreaker,
+                disableCircuitBreaker);
+
+        ChannelHandlerContext ctx = mock(ChannelHandlerContext.class);
+
+        // when
+        requestBuilderWrapper.setCtx(ctx);
+
+        // then
+        assertThat(ctx).isEqualTo(requestBuilderWrapper.getCtx());
+    }
+
+    @Test
+    public void get_set_Url_works_as_expected() {
```

Comment:
For this and the HTTP method get/set test, we should validate that it also calls the appropriate setter on the underlying `BoundRequestBuilder`. To do that, pass in a mock `BoundRequestBuilder` when creating the `RequestBuilderWrapper` and use `verify(requestBuilder).setThing(expectedValue)`.